lib/_http_agent.js

```diff
@@ -61,6 +61,7 @@ function Agent(options) {
   this.maxFreeSockets = this.options.maxFreeSockets || 256;
 
   this.on('free', (socket, options) => {
+    options = util._extend({}, options);
```

Comment:
The description is about `null` as option. In that case a default value would be better than copying the options if they are not manipulated else wise anywhere in the code. The same applies to the other entry.

Suggestion:
 
```js
if (options == null) options = {};
```

Reply:
I updated it @BridgeAR , review it please